### DISCUSSION TITLE
Moving agents

### DISCUSSION THREAD

**Discussion by hsp** [mesa/mesa#1154] [General]
I am new to Mesa. Not new to Python. Nor to ABM. 
Looking at the examples it appears that most of the models are based on cellular-automata style modelling, where agents are not moving, like flocks of birds or pedestrians. Is that observation correct? If not, please, can you provide references to such models?
Furthermore, are examples available where agents are moving along infrastructure networks - for example by application of shortest path's.
I do apologize for asking such a basic and open question.
/Hans

**Comment by aiwhoo** [comment]
One recent example that's being actively integrated into mesa is shown here: https://github.com/projectmesa/mesa/discussions/1147

You're right that most examples don't utilize the continuous space. The examples are lacking. The best example is boid flocker (https://github.com/projectmesa/mesa/tree/main/examples/boid_flockers); however, I feel the Continuous Space  has been neglected since the style is completely different than the other examples.

**Comment by rht** [comment]
OP didn't mention the requirement for modelling a continuous space. Movement is a bread and butter in a lot of the examples.
E.g.
- Epstein civil violence https://github.com/projectmesa/mesa/blob/c30ab5eef5ef0f62127cd4a71a9f8bc97df21678/examples/epstein_civil_violence/epstein_civil_violence/agent.py#L91-L93
- Schelling https://github.com/projectmesa/mesa/blob/c30ab5eef5ef0f62127cd4a71a9f8bc97df21678/examples/schelling/model.py#L33
- and so on

**Reply by aiwhoo** [reply]
Good point! I'm used to the continuous space so I incorrectly made an assumption. 

I think it holds though that we need more examples in the examples directory! =D

**Reply by rht** [reply]
The 2 examples I mentioned already served their purpose for showcasing movements.

**Comment by aiwhoo** [comment]
At the core, yes, this is understood by someone that really knows programming and modeling well; however, there’s ambiguity for beginners. Particularly in scenarios with defined path structures or following behaviors or desired destinations. With Mesa’s counterpart netlogo, there’s hundreds of coding examples that get beginners into it easily; while mesa only has 16 examples, which may not resonate and left the beginners  frustrated and returning back to other modeling platforms.

**Reply by tpike3** [reply]
@aiwhoo always happy to have more examples to merge.

On that note we are working with Santa Fe to set up a tutorial class through [complexity explorer](https://www.complexityexplorer.org/home) on Mesa. The plan is for second portion to walk through how to contribute an example (thanks @rht). 

@rht based on our other conversation my thought is to do sugar and space traders from sugarscape as the example...what do you think?

**Reply by rht** [reply]
> @rht based on our other conversation my thought is to do sugar and space traders from sugarscape as the example...what do you think?

Sounds good. You mentioned that those models depend on #1142 being merged. But #1142 has several design issues that have yet to be resolved. But there is always a workaround by just adding the by-type activation from scratch in the model's model.py.

**Comment by hsp** [comment]
Eventually a full integration of Geo-data and GIS/Spatial modelling capabilities would be great for creation of real-world models. As you probably all know there is a lot of very evolved options available. GDAL, OGR, Shapely etc. Inclusions of features (points, lines, and polygons), networks and raster’s would be good. Do you happen to know any examples?

**Comment by jackiekazil** [comment]
It already exists -- https://github.com/Corvince/mesa-geo
